src/NLog/Targets/FileTarget.cs

```diff
@@ -1058,22 +1052,21 @@ private void RollArchivesForward(string fileName, string pattern, int archiveNum
             string newFileName = ReplaceNumberPattern(pattern, archiveNumber);
             RollArchivesForward(newFileName, pattern, archiveNumber + 1);
 
-            var shouldCompress = archiveNumber == 0;
-            try
-            {
-                RollArchiveForward(fileName, newFileName, shouldCompress);
-            }
-            catch (IOException)
```

Comment:
I have one question: why has this changed to two method and why is the `catch` removed?